src/Controller.cpp

```diff
@@ -337,6 +337,7 @@ namespace geopm
         m_platform_io.read_batch();
         m_tracer->update(m_trace_sample);
         m_profile_tracer->update(m_application_sampler.get_records());
+        m_reporter->update();
```

Comment:
One other difference between these header/footer blocks and the `step()`'s `walk_up()` block:

```
        m_tracer->update(m_trace_sample);
        m_profile_tracer->update(m_application_sampler.get_records());
```
are called before `m_reporter->update()` here, but after the reporter inside `walk_up()`. Does that difference matter?

Reply:
It shouldn't matter, but i can make them consistent.

Reply:
done!

Reply:
Done now...